no `join_all` in `future`

**Describe the bug**
When compiling the git version of this repo, there are errors when it reaches matrix-sdk-crypto. No issues observed when using the crates released version.

**To Reproduce**
Steps to reproduce the behavior:
1. Add matrix-sdk to cargo.toml and reference the git link
2. cargo build

Output:

```
$ cargo build
    Blocking waiting for file lock on build directory
   Compiling bs58 v0.4.0
   Compiling anymap2 v0.13.0
   Compiling futures-task v0.3.18
   Compiling futures-channel v0.3.18
   Compiling async-stream v0.3.2
   Compiling ruma-identifiers-validation v0.5.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-serde v0.5.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling rqrr v0.4.0
   Compiling futures-util v0.3.18
   Compiling ruma-identifiers v0.20.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-common v0.6.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-api v0.18.5 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-signatures v0.9.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling matrix-qrcode v0.2.0 (https://github.com/matrix-org/matrix-rust-sdk#703e9651)
   Compiling h2 v0.3.9
   Compiling ruma-events v0.24.6 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling hyper v0.14.16
   Compiling hyper-tls v0.5.0
   Compiling reqwest v0.11.7
   Compiling ruma-state-res v0.4.1 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-client-api v0.12.3 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma-federation-api v0.3.1 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling ruma v0.4.0 (https://github.com/ruma/ruma/?rev=fdbc4d6d1dd273c8a6ac95b329943ed8c68df70d#fdbc4d6d)
   Compiling matrix-sdk-common v0.4.1 (https://github.com/matrix-org/matrix-rust-sdk#703e9651)
   Compiling matrix-sdk-crypto v0.4.1 (https://github.com/matrix-org/matrix-rust-sdk#703e9651)
error[E0432]: unresolved import `futures_util::future::join_all`
  --> /Users/lukepicciau/.cargo/git/checkouts/matrix-rust-sdk-1f4927f82a3d27bb/703e965/crates/matrix-sdk-crypto/src/identities/manager.rs:22:5
   |
22 | use futures_util::future::join_all;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `join_all` in `future`

error[E0432]: unresolved import `futures_util::future::join_all`
  --> /Users/lukepicciau/.cargo/git/checkouts/matrix-rust-sdk-1f4927f82a3d27bb/703e965/crates/matrix-sdk-crypto/src/session_manager/group_sessions.rs:22:5
   |
22 | use futures_util::future::join_all;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `join_all` in `future`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `matrix-sdk-crypto` due to 2 previous errors
```

Cargo.toml file:

```
[package]
name = "echobot"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
dotenv = "0.15.0"
tokio = {version = "1.14.0", features = ["full"]}
matrix-sdk = { git = "https://github.com/matrix-org/matrix-rust-sdk" }
```
matrix-sdk version: `source = "git+https://github.com/matrix-org/matrix-rust-sdk#703e965172584839f9bc7d5f359ef995fb0a226a"`

```
rustc --version
rustc 1.56.1 (59eed8a2a 2021-11-01)
```

--- a/crates/matrix-sdk-crypto/Cargo.toml
+++ b/crates/matrix-sdk-crypto/Cargo.toml
@@ -30,7 +30,7 @@ base64 = "0.13.0"
 bs58 = "0.4.0"
 byteorder = "1.4.3"
 dashmap = "4.0.2"
-futures-util = { version = "0.3.15", default-features = false }
+futures-util = { version = "0.3.15", default-features = false, features = ["alloc"] }
 getrandom = "0.2.3"
 hmac = "0.11.0"
 matrix-qrcode = { version = "0.2.0", path = "../matrix-qrcode", optional = true }

